zyte_spider_templates/spiders/base.py

```diff
@@ -1,85 +1,24 @@
-from enum import Enum
 from importlib.metadata import version
-from typing import Any, Dict, Optional
+from typing import Any, Dict
 
 import scrapy
-from pydantic import BaseModel, Field
+from pydantic import BaseModel
 from scrapy.crawler import Crawler
 
-from zyte_spider_templates._geolocations import (
-    GEOLOCATION_OPTIONS_WITH_CODE,
-    Geolocation,
-)
-from zyte_spider_templates.documentation import document_enum
+from ..params import ExtractFromParam, GeolocationParam, MaxRequestsParam, UrlParam
 
 # Higher priority than command-line-defined settings (40).
 ARG_SETTING_PRIORITY: int = 50
 
 
-@document_enum
-class ExtractFrom(str, Enum):
```

Comment:
Both `Geolocation` and `ExtractFrom` would previously be imported from here and are now imported from `zyte_spider_templates.params`.

I was hoping to deprecate them properly, but I had to go with a backward-incompatible change, because I could not think of a proper way of deprecating them that was not [rather hacky](https://stackoverflow.com/a/7668273/939364), given (TIL) `Enum` subclassing is not possible.